source/mir/random/ndvariable.d

```diff
@@ -279,6 +280,118 @@ nothrow @safe version(mir_random_test) unittest
     assert(fabs(x[0] + x[1] + x[2] - 1) < 1e-10);
 }
 
+/++
+Multinomial distribution.
++/
+struct MultinomialVariable(U:size_t, T)
+    if (isFloatingPoint!T)
+{
+    import mir.random.variable : binomialVar;
+
+    ///
+    enum isNdRandomVariable = true;
+    ///
+    alias Element = uint;
+
+    ///
+    const(T)[] probs;
+    size_t N;
+
+    /++
+    Params:
+        probs = probabilities of the multinomial distribution
+        N = Number of rolls
+    Constraints: `sum(probs[i]) <= 1`
+    +/
+    this()(size_t N, const(T)[] probs)
+    {
+        this.N = N;
+        this.probs = probs;
+    }
+
+    ///
+    pragma(inline, false)
+    void opCall(G)(scope ref G gen, scope U[] result)
+        if (isSaturatedRandomEngine!G)
+    {
+        auto K = this.probs.length;
+        double norm = 0.0;
+        double sum_p = 0.0;
+        alias rng = rne;
+
+        uint sum_n = 0;
+
+
+        /// Makes sure probabilities add up to one, by calculating a normalization term
```

Comment:
Why not use `auto norm = sum(this.probs);`? Then you can do something like `typeof(norm) sum_p = 0;`.

Reply:
to have `sum()` I need to import it from mir-ndslice. is it worth it?

Reply:
I would leave that up to @9il

Reply:
`sum` is from `mir-algorithm` that is an optional dependency. So, it shouldn't be imported in this module.